db/migrations/000010_add_fields_to_environments.up.sql

```diff
@@ -0,0 +1,12 @@
+ALTER TABLE environments
+ADD COLUMN destroyed_at timestamp WITH TIME ZONE,
+ADD COLUMN is_permanent boolean,
+ADD COLUMN requester text,
+ADD COLUMN namespace text,
+ADD COLUMN customization jsonb,
```

Comment:
that's for the customizations per preview environment, like the unique name, specfic chart versions for apps.

Reply:
Is an unstructured json blob the right way to handler that type of thing? 🤔

Reply:
ah gotcha! Thanks.

In the spirit of offering EVEN MORE unsolicited opinions I wonder if there's merit in formally modeling this in the DB and API rather than allowing unstructured data. Just because:
* we're less likely to end up with bad data in the db from a misconfigured client
* there's a well-defined flow for making schema changes

Reply:
I'm totally pro structured-api! I don't want people just throw blobs into our database, I'm fully expecting this column to be some sort of a rendered json created by us via api parameters or something we control.

I'm hesitant to formally model columns for a couple scenarios
a) I'm expecting the api to control/sanitize the input into the db, so I'm expecting that to deal with client-sourced problems.
b) if there's a new chart that gets added to helm, we would have to add a new row to give it a version, at the same time, deprecated services will end up with permanently dead rows. db migrations are scary and not something I'd like to do often or under external pressure (oh hey we have a new service, can you support it for us?).
c) I'm not clear atm what the scope of various customizations are for environments, essentially they're just helm charts overriding base configs afaik, so we might find this-or-that to be useful to tweak for various reasons (i dunno having a perf test have extra pods). Having to continuously think about db migrations/rollbacks every time we want to try something would just make it really hard.


Sorry if I'm misunderstanding something!